core/modules/rproc/InfileMerger.cc

```diff
@@ -350,51 +350,42 @@ int InfileMerger::makeJobIdAttempt(int jobId, int attemptCount) {
 }
 
 
-sql::Schema InfileMerger::getSchemaForQueryResults(query::SelectStmt const& stmt, std::string& errMsg) {
+bool InfileMerger::getSchemaForQueryResults(query::SelectStmt const& stmt, sql::Schema& schema) {
     sql::SqlResults results;
     sql::SqlErrorObject getSchemaErrObj;
     std::string query = stmt.getQueryTemplate().sqlFragment();
     bool ok = _applySqlLocal(query, results, getSchemaErrObj);
     if (not ok) {
         LOGS(_log, LOG_LVL_ERROR, "Failed to get schema:" << getSchemaErrObj.errMsg());
-        errMsg = getSchemaErrObj.errMsg();
-        return sql::Schema();
+        _error = InfileMergerError(util::ErrorCode::INTERNAL, "Failed to get schema: " + getSchemaErrObj.errMsg());
+        return false;
     }
     sql::SqlErrorObject errObj;
     if (errObj.isSet()) {
-        LOGS(_log, LOG_LVL_ERROR, "failed to extract schema from result: " << errObj.errMsg());
-        errMsg = errObj.errMsg();
-        return sql::Schema();
+        LOGS(_log, LOG_LVL_ERROR, "Failed to extract schema from result: " << errObj.errMsg());
+        _error = InfileMergerError(util::ErrorCode::INTERNAL, "Failed to extract schema from result: " + errObj.errMsg());
+        return false;
     }
-    auto schema = results.makeSchema(errObj);
+    schema = results.makeSchema(errObj);
     LOGS(_log, LOG_LVL_DEBUG, "InfileMerger extracted schema: " << schema);
-    return schema;
+    return true;
```

Comment:
Next logical step would be to make it throw an exception (and return schema), that would make reviewers even happier. 😸

Reply:
I'm all with you, but this is where you already dinged me once for throwing where the rest of the class was return-resulting. I agree, it wants a wholesale refactor, but not today 🙃